### DISCUSSION TITLE
Loop for popup of content preferences and location details

### DISCUSSION THREAD

**Discussion by raghupathiguduri** [Sunbird-Ed/Community#456] [Sunbird ED configuration discussion] [answered]
Hi Team, 
we have installed 4.6.0 version of sunbird. And we didn't enable signup option from login page, so we are creating the users through api. 
After creating API when the user tries to login for the first time we are having issue with popup of location details and content preferences.

steps to reproduce
1.Create a new user using API and log in with it
2.Accept the terms and conditions
3.Submit all the mandatory fields from popups
4.The popup for board medium class and subject keeps appearing again and again until refreshed

check the video for reference
[Pop up issues_compressed.webm](https://user-images.githubusercontent.com/48198006/231151383-10e939cc-724e-4dc5-982b-7e86077b348c.webm)

Also we have another issue. 
When we login and try to change the preferences it is throwing error message. when we check the network console , it is calling an undefined channel read call but for the second time it is taking the channel id from search call.
![image (8)](https://user-images.githubusercontent.com/48198006/231154741-5ef49fa2-1755-4d84-a69a-76db8dc78a4b.png)
![image (7)](https://user-images.githubusercontent.com/48198006/231154746-cfb53ec2-73ef-47cf-ab0e-a050d1c741db.png)

**Comment by rajeevsathish** [comment]
@raghupathiguduri for the org that you create user is the custodianOrg and that is 0136095267023831049 
there is api call 
`curl --location 'https://sunbirdsaas.com/api/channel/v1/read/0136095267023831049' \
--header 'Accept: application/json' \
--header 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'Connection: keep-alive' \
--header 'Cookie: connect.sid=s%3AYVwLpgPuRlqtwiotZ0QHwOJFAOyjHGep.CLgaJvPW6pH4UiYEdYs2VgtFlGwZzm%2FECiDyDDsh7wI' \
--header 'Referer: https://sunbirdsaas.com/explore' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36' \
--header 'X-App-Id: production.production.portal' \
--header 'X-App-Version: 4.6.0' \
--header 'X-Channel-Id: 0124487522476933120' \
--header 'X-Device-ID: eaf3c94e769d7dad4cd73075e9208117' \
--header 'X-Org-code: 0124487522476933120' \
--header 'X-Request-ID: 8bc91693-cea2-6657-fa60-bea025944802' \
--header 'X-Session-ID: d914612a-57e8-b60b-feb0-a6699d8b9681' \
--header 'X-Source: web' \
--header 'X-msgid: 8bc91693-cea2-6657-fa60-bea025944802' \
--header 'sec-ch-ua: "Google Chrome";v="111", "Not(A:Brand";v="8", "Chromium";v="111"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "macOS"' \
--header 'ts: 2023-04-12T12:45:21+05:30'`

response is  
`{
    "id": "api.channel.read",
    "ver": "1.0",
    "ts": "2023-04-12T07:15:21.169Z",
    "params": {
        "resmsgid": "c92c9410-d901-11ed-af0e-55c5d99ca9ba",
        "msgid": "8bc91693-cea2-6657-fa60-bea025944802",
        "status": "successful",
        "err": null,
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "channel": {
            "identifier": "0136095267023831049",
            "lastStatusChangedOn": "2022-08-24T05:28:45.863+0000",
            "code": "0136095267023831049",
            "consumerId": "7d9e3ca8-600e-448d-a9d9-0038238d78fa",
            "channel": "in.ekstep",
            "description": "Default Organisation for Custodian",
            "assetAdditionalCategories": [],
            "languageCode": [],
            "suggested_frameworks": [
                {
                    "identifier": "andhrapradesh",
                    "code": "andhrapradesh",
                    "name": "Andhra Pradesh",
                    "objectType": "Framework"
                },
                {
                    "identifier": "andhrapradeshfw",
                    "code": "andhrapradeshfw",
                    "name": "Andhra Pradesh",
                    "objectType": "Framework"
                },
                {
                    "identifier": "chattisgarghboardfw",
                    "code": "chattisgarghboardfw",
                    "name": "Chattisgargh  Board of Secondary Education",
                    "objectType": "Framework"
                },
                {
                    "identifier": "climatechange1",
                    "code": "climatechange1",
                    "name": "climatechange1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "devsandboxfw",
                    "code": "devsandboxfw",
                    "name": "Dev Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "devsandbox1fw",
                    "code": "devsandbox1fw",
                    "name": "Dev Framework1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "devsandbox2fw",
                    "code": "devsandbox2fw",
                    "name": "Dev Framework2",
                    "objectType": "Framework"
                },
                {
                    "identifier": "devsandbox3fw",
                    "code": "devsandbox3fw",
                    "name": "Dev Framework3",
                    "objectType": "Framework"
                },
                {
                    "identifier": "dhwanifoundationfw",
                    "code": "dhwanifoundationfw",
                    "name": "Dhwani Foundation",
                    "objectType": "Framework"
                },
                {
                    "identifier": "dhwanifoundfw",
                    "code": "dhwanifoundfw",
                    "name": "Dhwani Foundation",
                    "objectType": "Framework"
                },
                {
                    "identifier": "dummysandboxfw3",
                    "code": "dummysandboxfw3",
                    "name": "Dummy Test Framework I",
                    "objectType": "Framework"
                },
                {
                    "identifier": "dummysandboxfw2",
                    "code": "dummysandboxfw2",
                    "name": "Dummy Test Framework II",
                    "objectType": "Framework"
                },
                {
                    "identifier": "egmodulesforagsfw",
                    "code": "egmodulesforagsfw",
                    "name": "EG Modules for AGs",
                    "objectType": "Framework"
                },
                {
                    "identifier": "egmodulesforags",
                    "code": "egmodulesforags",
                    "name": "EG Modules for AGs",
                    "objectType": "Framework"
                },
                {
                    "identifier": "gujaratboardfw",
                    "code": "gujaratboardfw",
                    "name": "Gujarat Secondary and Higher Secondary Education Board",
                    "objectType": "Framework"
                },
                {
                    "identifier": "hindifw",
                    "code": "hindifw",
                    "name": "Hindi",
                    "objectType": "Framework"
                },
                {
                    "identifier": "indeddemofw",
                    "code": "indeddemofw",
                    "name": "INDed_Demo",
                    "objectType": "Framework"
                },
                {
                    "identifier": "INDedPreview",
                    "code": "INDedPreview",
                    "name": "INDedPreview",
                    "objectType": "Framework"
                },
                {
                    "identifier": "india",
                    "code": "india",
                    "name": "India",
                    "objectType": "Framework"
                },
                {
                    "identifier": "keyeducationfoundationfw",
                    "code": "keyeducationfoundationfw",
                    "name": "Key Education Foundation",
                    "objectType": "Framework"
                },
                {
                    "identifier": "keyeducationfoundation",
                    "code": "keyeducationfoundation",
                    "name": "Key Education Foundation",
                    "objectType": "Framework"
                },
                {
                    "identifier": "mission100fw",
                    "code": "mission100fw",
                    "name": "Mission 100 Project Training Content framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "NCF",
                    "code": "NCF",
                    "name": "NCF framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "nidhisandboxfw2",
                    "code": "nidhisandboxfw2",
                    "name": "Nidhi Sandbox Framework I",
                    "objectType": "Framework"
                },
                {
                    "identifier": "nidhisandboxfw3",
                    "code": "nidhisandboxfw3",
                    "name": "Nidhi Sandbox Framework II",
                    "objectType": "Framework"
                },
                {
                    "identifier": "niosfw",
                    "code": "niosfw",
                    "name": "NIOS",
                    "objectType": "Framework"
                },
                {
                    "identifier": "nios",
                    "code": "nios",
                    "name": "NIOS",
                    "objectType": "Framework"
                },
                {
                    "identifier": "odisha",
                    "code": "odisha",
                    "name": "Odisha",
                    "objectType": "Framework"
                },
                {
                    "identifier": "odishafw",
                    "code": "odishafw",
                    "name": "Odisha",
                    "objectType": "Framework"
                },
                {
                    "identifier": "defaultboard",
                    "code": "defaultboard",
                    "name": "People Act Sandbox Framework default board 1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "pplactsandboxfw",
                    "code": "pplactsandboxfw",
                    "name": "People Action Sandbox Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "pplactsandboxfw2",
                    "code": "pplactsandboxfw2",
                    "name": "People Action Sandbox Framework II",
                    "objectType": "Framework"
                },
                {
                    "identifier": "prerakscorner",
                    "code": "prerakscorner",
                    "name": "Prerak Corner",
                    "objectType": "Framework"
                },
                {
                    "identifier": "pragatifw",
                    "code": "pragatifw",
                    "name": "Prerak's Corner",
                    "objectType": "Framework"
                },
                {
                    "identifier": "rsosfw",
                    "code": "rsosfw",
                    "name": "RSOS",
                    "objectType": "Framework"
                },
                {
                    "identifier": "rsos",
                    "code": "rsos",
                    "name": "RSOS",
                    "objectType": "Framework"
                },
                {
                    "identifier": "eglsandboxfw",
                    "code": "eglsandboxfw",
                    "name": "RSOS",
                    "objectType": "Framework"
                },
                {
                    "identifier": "sandboxfw",
                    "code": "sandboxfw",
                    "name": "Sandbox framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "shrutifw",
                    "code": "shrutifw",
                    "name": "Shruti_Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tan90fw",
                    "code": "tan90fw",
                    "name": "STEM Training ",
                    "objectType": "Framework"
                },
                {
                    "identifier": "sunbirdlearning",
                    "code": "sunbirdlearning",
                    "name": "Sunbird Learning",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tnflnfw",
                    "code": "tnflnfw",
                    "name": "Tamilnadu-FLN Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "techjoomlashruti",
                    "code": "techjoomlashruti",
                    "name": "Techjoomla Shruti Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "techjoomlafw",
                    "code": "techjoomlafw",
                    "name": "techjoomlafw",
                    "objectType": "Framework"
                },
                {
                    "identifier": "techjoomlafw1",
                    "code": "techjoomlafw1",
                    "name": "techjoomlafw1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "techjoomlafw2",
                    "code": "techjoomlafw2",
                    "name": "techjoomlafw2",
                    "objectType": "Framework"
                },
                {
                    "identifier": "ttplacademy",
                    "code": "ttplacademy",
                    "name": "Tekdi Academy",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tekdidemosandboxfw",
                    "code": "tekdidemosandboxfw",
                    "name": "Tekdi demo sandbox framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tekdisandboxfw",
                    "code": "tekdisandboxfw",
                    "name": "Tekdi SandBox Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tekditestSachinframework",
                    "code": "tekditestSachinframework",
                    "name": "Tekdi Test Sachin Framework",
                    "objectType": "Framework"
                },
                {
                    "identifier": "tekditech",
                    "code": "tekditech",
                    "name": "tekditech",
                    "objectType": "Framework"
                },
                {
                    "identifier": "telanganaa",
                    "code": "telanganaa",
                    "name": "Telangana",
                    "objectType": "Framework"
                },
                {
                    "identifier": "telangana",
                    "code": "telangana",
                    "name": "Telangana",
                    "objectType": "Framework"
                },
                {
                    "identifier": "ttplsandboxfw2",
                    "code": "ttplsandboxfw2",
                    "name": "TTPL Board",
                    "objectType": "Framework"
                },
                {
                    "identifier": "udaipurboard",
                    "code": "udaipurboard",
                    "name": "Udaipur Board",
                    "objectType": "Framework"
                },
                {
                    "identifier": "climatechange",
                    "code": "climatechange",
                    "name": "VKS Framework 1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "vks2023A",
                    "code": "vks2023A",
                    "name": "vks2023A",
                    "objectType": "Framework"
                },
                {
                    "identifier": "vksfw100",
                    "code": "vksfw100",
                    "name": "vksfw100",
                    "objectType": "Framework"
                },
                {
                    "identifier": "vksfw2",
                    "code": "vksfw2",
                    "name": "vksfw2",
                    "objectType": "Framework"
                },
                {
                    "identifier": "vkstest1",
                    "code": "vkstest1",
                    "name": "vkstest1",
                    "objectType": "Framework"
                },
                {
                    "identifier": "waterabc10",
                    "code": "waterabc10",
                    "name": "waterabc10",
                    "objectType": "Framework"
                },
                {
                    "identifier": "waterabc12",
                    "code": "waterabc12",
                    "name": "waterabc12",
                    "objectType": "Framework"
                },
                {
                    "identifier": "waterabc13",
                    "code": "waterabc13",
                    "name": "waterabc13",
                    "objectType": "Framework"
                }
            ],
            "createdOn": "2022-08-24T05:28:45.863+0000",
            "objectType": "Channel",
            "versionKey": "1661318925863",
            "collectionPrimaryCategories": [
                "Content Playlist",
                "Course",
                "Digital Textbook",
                "Question paper"
            ],
            "contentPrimaryCategories": [
                "Course Assessment",
                "eTextbook",
                "Explanation Content",
                "Learning Resource",
                "Practice Question Set",
                "Teacher Resource",
                "Exam Question"
            ],
            "appId": "slecosystem.slecosystem.portal",
            "name": "shikshalokamCustodian",
            "primaryCategories": [
                {
                    "identifier": "obj-cat:practice-question-set_questionset_all",
                    "name": "Practice Question Set",
                    "targetObjectType": "QuestionSet"
                }
            ],
            "additionalCategories": [],
            "lastUpdatedOn": "2022-08-24T05:28:45.863+0000",
            "collectionAdditionalCategories": [
                "Textbook",
                "Lesson Plan"
            ],
            "assetPrimaryCategories": [
                "Asset",
                "CertAsset",
                "Certificate Template"
            ],
            "contentAdditionalCategories": [
                "Classroom Teaching Video",
                "Concept Map",
                "Curiosity Question Set",
                "Experiential Resource",
                "Explanation Video",
                "Focus Spot",
                "Learning Outcome Definition",
                "Lesson Plan",
                "Marking Scheme Rubric",
                "Pedagogy Flow",
                "Previous Board Exam Papers",
                "TV Lesson",
                "Textbook"
            ],
            "status": "Live",
            "defaultFramework": "NCF"
        }
    }
}`

this has only one framework NCF

**Reply by chitranshu-keshav** [reply]
@raghupathiguduri Can you mark this as answered if the above resolves your issue. If not then please get back with details

**Reply by raghupathiguduri** [reply]
@rajeevsathish Thank you for the answer, But I want to understand that do we need to create one more framework to fix this issue ? or what should be the resolution ?

**Comment by Krishnaj20** [comment]
Marking this as answered since we haven't received any response.

**Comment by rajeevsathish** [comment, marked as answer]
@raghupathiguduri you can add the framework array for the channel and that should fix the issue.
